### ISSUE TITLE
[lit-html] Template literals with parts inside `<template>`s render incorrectly.

### ISSUE THREAD

**Issue by bicknellr** [lit/lit#1737]
# Description

When rendering a template string where some of the parts are inside of a `<template>`, values are rendered into the wrong parts. ([`TemplateInstance#clone()`](https://github.com/Polymer/lit-html/blob/2be1fcf0a82a2bf368295d015613cd87c5ad96f6/packages/lit-html/src/lit-html.ts#L750) doesn't walk `<template>` content to find parts.)

# lit-html version

2.0.0-pre.7

# Steps to Reproduce

1. Write this code

```ts
import {html, render} from 'https://unpkg.com/lit-html@2.0.0-pre.7?module';
render(html`${'A'}<template>${'B'}</template>${'C'}`, document.body);
```

2. See this output...

```
AB
```

## Live Reproduction Link
https://jsbin.com/xojudijari/edit?html,output

## Expected Results
The page shows `AC`. (`'B'` should be rendered to a part inside a `<template>`.)

## Actual Results
The page shows `AB`.

**Comment by stefanholzapfel** [comment]
I guess this is also the reason why directives in templates don't work anymore? 

If yes, that breaks e.g. <vaadin-select> for my use case, because I want to translate the options wit a directive.

**Comment by stefanholzapfel** [comment]
Marcus hellberg clarified that I was using vaadin-select in an old fashioned way. With the new way (without directly using templates: https://vaadin.com/docs/latest/ds/components/select) it works. Nevertheless there's an issue with directives / parts in templates.

**Comment by Artur-** [comment]
I guess I am seeing this same issue when using `<vaadin-select>`

Doing something like
```
  render() {
    const font = 'Cambria';
    return html`
      Hello
      <vaadin-select id="font-family" @change="${() => window.alert('hehllo')}">
        <template>
          <vaadin-list-box>
            <vaadin-item value="">System default</vaadin-item>
            <vaadin-item style="font-family: ${font}" value="${font}"
              >${font}</vaadin-item
            >
          </vaadin-list-box>
        </template>
      </vaadin-select>
      <vaadin-radio-group @change="${() => window.alert('hello')}">
      </vaadin-radio-group>
    `;
  }
```

results in nothing rendered and
```
Uncaught (in promise) TypeError: Cannot read property 'split' of null
```
originating in https://github.com/lit/lit/blob/main/packages/lit-html/src/lit-html.ts#L613 where `realName` is something completely different than what the code expects. The element is `<vaadin-radio-group>` and `realName` is `style` which is not present on that element

**Comment by sorvell** [comment]
This can be done using the `static-html` module, as shown in this [example](https://lit.dev/playground/#project=W3sibmFtZSI6InNpbXBsZS1ncmVldGluZy50cyIsImNvbnRlbnQiOiJpbXBvcnQge2NzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7aHRtbCwgdW5zYWZlU3RhdGljfSBmcm9tICdsaXQvc3RhdGljLWh0bWwuanMnO1xuaW1wb3J0IHtjdXN0b21FbGVtZW50LCBwcm9wZXJ0eSwgcXVlcnl9IGZyb20gJ2xpdC9kZWNvcmF0b3JzLmpzJztcblxuQGN1c3RvbUVsZW1lbnQoJ3NpbXBsZS1ncmVldGluZycpXG5leHBvcnQgY2xhc3MgU2ltcGxlR3JlZXRpbmcgZXh0ZW5kcyBMaXRFbGVtZW50IHtcbiAgc3RhdGljIHN0eWxlcyA9IGNzc2BwIHsgY29sb3I6IGJsdWUgfWA7XG5cbiAgQHByb3BlcnR5KClcbiAgbmFtZSA9ICdTb21lYm9keSc7XG4gIFxuICBAcXVlcnkoJyN0ZW1wbGF0ZScpXG4gIHRlbXBsYXRlPzogSFRNTFRlbXBsYXRlRWxlbWVudDtcbiAgXG4gIEBxdWVyeSgnI2NvbnRhaW5lcicpXG4gIGNvbnRhaW5lcj86IEhUTUxEaXZFbGVtZW50O1xuXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGA8cD5IZWxsbywgJHt0aGlzLm5hbWV9ITwvcD5cbiAgICAgICAgPHRlbXBsYXRlIGlkPVwidGVtcGxhdGVcIj5IZWxsbywgJHt1bnNhZmVTdGF0aWModGhpcy5uYW1lKX08L3RlbXBsYXRlPlxuICAgICAgICA8ZGl2IGlkPVwiY29udGFpbmVyXCI-PC9kaXY-YDtcbiAgfVxuICB1cGRhdGVkKCkge1xuICAgIHRoaXMuY29udGFpbmVyLmlubmVySFRNTCA9ICcnO1xuICAgIHRoaXMuY29udGFpbmVyLmFwcGVuZENoaWxkKHRoaXMudGVtcGxhdGUuY29udGVudC5jbG9uZU5vZGUodHJ1ZSkpO1xuICB9XG59XG4ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPGhlYWQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vc2ltcGxlLWdyZWV0aW5nLmpzXCI-PC9zY3JpcHQ-XG48L2hlYWQ-XG48Ym9keT5cbiAgPHNpbXBsZS1ncmVldGluZyBuYW1lPVwiV29ybGRcIj48L3NpbXBsZS1ncmVldGluZz5cbjwvYm9keT5cbiJ9LHsibmFtZSI6InBhY2thZ2UuanNvbiIsImNvbnRlbnQiOiJ7XCJkZXBlbmRlbmNpZXNcIjp7XCJsaXRcIjpcIl4yLjAuMC1yYy4yXCIsXCJAbGl0L3JlYWN0aXZlLWVsZW1lbnRcIjpcIl4xLjAuMC1yYy4yXCIsXCJsaXQtZWxlbWVudFwiOlwiXjMuMC4wLXJjLjJcIixcImxpdC1odG1sXCI6XCJeMi4wLjAtcmMuM1wifX0ifV0).

If the values change, the entire template here will be re-created so this pattern should only be used for mostly static content.

**Comment by justinfagnani** [comment]
#1759 was too expensive to merge. We'd like to add a dev mode warning for bindings in `<template>` tags and either recommend using static templates or vend a new directive for putting bindings in `<template>`s.

**Comment by sorvell** [comment]
Addressed via #2114
